File: dcp-solutions/2024-09-26/solution.go
```go
package main

import (
	"fmt"
	"strings"
)

func longestAbsolutePath(s string) int {
	stack := []string{}
	res := 0

	for _, v := range strings.Split(s, "\n") {
		if len(v) == 0 {
			continue
		}

		v = strings.TrimLeft(v, "\t")
		if v[0] == '=' {
			for len(stack) > 1 && stack[len(stack)-1][0] != '=' {
				res = max(res, len(strings.Join(stack[:len(stack)-1], "/")) + 1 + len(v))
				stack = stack[:len(stack)-1]
			}
			if len(stack) == 1 {
				res = max(res, len(stack[0]) + len(v) + 1)
			} else {
				res = max(res, len(strings.Join(stack, "/")) + 1 + len(v))
			}
			stack = stack[:0]
		} else if v != "" && v[0] != '/' {
			stack = append(stack, v)
		}
	}

	return res
}

func max(a, b int) int {
	if a > b {
		return a
	}
	return b
}

func main() {

}
```

File: dcp-solutions/2024-09-26/solution.ex
```elixir
def longest_absolute_path(file_system) do
  file_system
  |> String.split("\n")
  |> Enum.map(&parse_directory/1)
  |> Enum.map(&file_path/1)
  |> Enum.max(&compare_paths/2)

  defp parse_directory(directory) do
    lines = directory
      |> String.split("\t")
      |> Enum.map(&String.trim/1)

    subdirectories = []
    files = []

    for line <- lines do
      case line do
        "=" <> _ -> :file => [line, []]
        _ when not String.starts_with?(line, " ") ->
          file_name = String.split(line, "/"). |> Enum.at(-1)
          subdirectories: [line] ++ Enum.map(&parse_directory/1, &line/1),
            files: [file_name => line]
      end
    end

    case {subdirectories, files} do
      {[_, _ | _], [file_name: ^file_name, file_content: _]} ->
        [file: [^file_name, file_content]]
      {[subdirectories, files], [subdirectories: subdirectory, files: [file_name => line]]} when is_bitwise_and(subdirectory, 2) -> [
          dir: [subdirectory, ^file_name],
          subdir: [dir: [subdirectory, ^file_name], subdir: [], files: [^file_name]]
        ]
      {[_, _ | _], []} -> []
    end
  end

  defp file_path({:file, [name, content]}), do: [name]
  defp file_path(:dir), do: []
  defp file_path({:subdir, [dir, name, subdirs]}), do: [dir] ++ file_path(subdirs)
end

defmodule BitwiseAnd do
  @enforce_keys [:integer]

  @doc false
  defmacro __sizeof__(%{integer: value}) when is_integer(value) do
    value
  end

  defstruct(integer: ~w(integer))

  def new(integer), do: %{integer: integer}
end

defmodule EnumUtils do
  @spec all/1
  def all(list), do: List.all(list, fn elem -> true end)

  @spec any/1
  def any(list), do: List.any?(list, fn elem -> true end)
end